### DISCUSSION TITLE
Router back function not working.

### DISCUSSION THREAD

**Discussion by nicocarballo21** [vercel/next.js#67109] [Help]
### Summary

Its pretty simple, i have a list of tokens where each token has a Link from `next/Link` that redirect the user to /token/:tokenId.

```js
 <Link className={className} href={`/token/${token.id}`}>
```
On that /token/:tokenId page, there is a button that executes `router.back()` from `useRouter()` from `next/router`

```js
 onClick={() => router.back()}
```

The problem with this is:

- I can go to the dynamic page without problems but when i try to go back to the main page.
- I hit the button, the `url` on the browser change to `/` which is correct, but the page dont make a render again and it gets stuck on the dynamic page instead go back to the home page.

CC: This happens the first time, once you start browsing the site, for some reason it resolves itself.

### Additional information

```js
NextJS version: 14.2.4
Errors: None
Usign pages, not app router
```

**Comment by Mnigos** [comment]
I got same issue

**Comment by anhtuan219** [comment]
+1
I have the same issue using the pages router.

Temporarily, here's a dumb way to get around this
```
onClick={() => {
  router.back();
  setTimeout(router.reload, 0);  // ---> reload page after go back history
}}
```

Hope the vercel team will fix this issue soon
